### PR TITLE
Feature/gm 26 equation math checker

### PR DESCRIPTION
**Pull request by Morshall2508**
Implemented, for all test returns passed.

### CODE REVIEW

File: src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                return false;
+
+        } else if (equation.contains("-")) {
+            if (a - b == c)
+                return true;
+
+            if (a - b != c)
+                return false;
+        }
+        throw new RuntimeException("Not implemented");
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java:29]: I think now it is implemented ;)

File: src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+@Component
+class EquationMathChecker {
+
+
+    boolean isMathematicallyCorrect(String equation) {
+
+        char[] equationChar = equation.toCharArray();
+
+        int a = Integer.parseInt(String.valueOf(equationChar[0]));
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java:13]: `int a = equationChar[0] - '0';` is another option

File: src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        int b = Integer.parseInt(String.valueOf(equationChar[2]));
+        int c = Integer.parseInt(String.valueOf(equationChar[4]));
+        if (equation.contains("+")) {
+            if (a + b == c)
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java:17]:
```
if (a + b == c)
    return true;
if (a + b != c)
    return false;
```

can be simplified with

`a+b==c`

similar with `-`

File: src/test/java/pl/piekoszek/gorskimatches/equation/EquationMathCheckerTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        assertThat(equationMathChecker.isMathematicallyCorrect("0-0=0")).isTrue();
+        assertThat(equationMathChecker.isMathematicallyCorrect("5-2=3")).isTrue();
+        assertThat(equationMathChecker.isMathematicallyCorrect("4-1=3")).isTrue();
+        assertThat(equationMathChecker.isMathematicallyCorrect("9-9=0")).isTrue();
+        assertThat(equationMathChecker.isMathematicallyCorrect("4-4=0")).isTrue();
+        assertThat(equationMathChecker.isMathematicallyCorrect("7-2=5")).isTrue();
+    }
+
+    @Test
+    void shouldReturnFalseOnIncorrectAddEquations() {
```
Comment (piekaa) [inline src/test/java/pl/piekoszek/gorskimatches/equation/EquationMathCheckerTest.java:32]: You can add 2+2=0 test case

**Review by piekaa — CHANGES_REQUESTED** [review]
Some simplifications can be done

File: src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java (hunk 2 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java:13]: When changed to that - id does not pass test.

File: src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java (hunk 3 above)
Comment (Morshall2508) [reply src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java:17]: ok

File: src/test/java/pl/piekoszek/gorskimatches/equation/EquationMathCheckerTest.java (hunk 4 above)
Comment (Morshall2508) [reply src/test/java/pl/piekoszek/gorskimatches/equation/EquationMathCheckerTest.java:32]: ok

File: src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    boolean isMathematicallyCorrect(String equation) {
+
+        char[] equationChar = equation.toCharArray();
+
+        int a = Integer.parseInt(String.valueOf(equationChar[0]));
+        int b = Integer.parseInt(String.valueOf(equationChar[2]));
+        int c = Integer.parseInt(String.valueOf(equationChar[4]));
+
+        if (equation.contains("+")) {
+            if (a + b == c) {
```
Comment (piekaa) [inline src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java:18]:
```
if (a + b == c) {
    return true;
}
```
can be simplified to 
return `a + b == c`

File: src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java (hunk 2 above)
Comment (piekaa) [reply src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java:13]: How did you change it?

**Review by piekaa — APPROVED** [review]
Good work ;)

**Pull request merged**